### PR TITLE
US87822 - Populate all assignments view

### PR DESCRIPTION
**Pull request by apalaniuk**
https://rally1.rallydev.com/#/128188728532d/detail/userstory/128262149148

Creates the "All Upcoming Activities" view and populates it. There are some minor(ish?) styling issues that I think can come in a second pass to allow all related dependencies/PRs to get in first.

Not yet covered: 
* "Complete"/"Due Today" indicators in overlay - ~will be added when #16 is merged, and it's cleared up whether the design is the same for both mobile and desktop overlay views~ Or, probably in a separate PR
* Ellipsis on instructions overflow: Cross-browser ellipsis truncation in CSS isn't a thing - will evaluate some other solutions/what's acceptable in the future polish PR

Requires: 
https://github.com/Brightspace/parent-portal-ui/pull/239
https://git.dev.d2l/projects/IE/repos/parentportal/pull-requests/124/overview

### CODE REVIEW

File: components/d2l-all-assessments-list.html [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+				display: block;
+				width: 70%;
+				min-width: 300px;
+				margin: auto;
+			}
+
+			.date-header {
+				@apply --d2l-body-compact-text;
+				font-weight: 300;
+				font-size: 0.7rem;
```
Comment (CodeBaboon) [inline components/d2l-all-assessments-list.html]: I think there needs to maybe be a `margin-top: 40px` or something equivalent on all `.date-header`s after the first one.

File: components/d2l-all-assessments-list.html [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+<link rel="import" href="../src/behaviors/date-behavior.html">
+<link rel="import" href="../src/behaviors/localize-behavior.html">
+<link rel="import" href="d2l-all-assessments-list-item.html">
+
+<dom-module id="d2l-all-assessments-list">
+	<template>
+		<style>
+			:host {
+				display: block;
+				width: 70%;
```
Comment (CodeBaboon) [inline components/d2l-all-assessments-list.html]: This seems to grow a bit too wide at max desktop width 1230px. Mock appears to be about `65%` at that width.

File: components/d2l-all-assessments-list-item.html [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+<dom-module id="d2l-all-assessments-list-item">
+	<template>
+		<style>
+			:host {
+				display: block;
+				margin: 10px auto;
+				background-color: var(--d2l-color-white);
+				border: 1px var(--d2l-color-titanius) solid;
+				border-radius: 6px;
```
Comment (CodeBaboon) [inline components/d2l-all-assessments-list-item.html]: This appears to be more rounded than the mock, to my eyes `4px;` appears to be a closer match, but if these values are specified somewhere explicitly or are consistent across other areas or something I'm cool with them.

Comment (apalaniuk) [inline components/d2l-all-assessments-list-item.html]: Nah, I think you're right - these are based on direct comparison + ruler extensions + using similar styling on other components, but are probably off. I'll note all of these and go over them with Bob at the same time for exact values.

File: components/d2l-all-assessments-list.html (hunk 1 above)
Comment (apalaniuk) [inline components/d2l-all-assessments-list.html]: Yeah, I also noticed that multiple activities on the same date weren't stacking correctly, so I'll lump that together in the next pass.

**Pull request merged**